### ISSUE TITLE
ILLinker should special case facade assemblies

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/runtime#90469]
### Description

See https://github.com/dotnet/runtime/issues/82169

There are known assemblies that has some reference assemblies that may not be existing. The case I was seeing is `System.Runtime.Serialization -> System.Runtime.Serialization.Schema`.

When **not** using `--skip-unresolved` flag of the linker, it reports IL1009 error since it couldn't resolve `System.Runtime.Serialization.Schema`.

### Reproduction Steps

I don't have an isolated repro right now. But the case was exactly that `System.Runtime.Serialization.Schema` can't be resolved.

### Expected behavior

As https://github.com/dotnet/runtime/issues/82169#issuecomment-1431658226 outlines:

Basically, those specific handful of libraries should be treated as special and not resolved for dependency analysis.

### Actual behavior

IL1009

**Comment by vitek-karas** [comment]
This is exactly what the `--skip-unresolved` flag is for. There are lot of assemblies which has references to code which may not be part of the app - both in the framework but also in nugets and everywhere else. Special casing some assemblies would not really solve the problem. In the framework this most frequently happens with facade assemblies, but outside of framework facade assemblies are very rare, but the same issue still exists. I don't know of a good way to detect assemblies like that- other than they have unresolvable reference.

I think a better argument would be to make `--skip-unresolved` the default behavior even on the command line - it already is the default when running the trimmer from the SDK. But since SDK already does this, the priority of changing the command line behavior is pretty low.

Curious: are you running `illink` outside of the SDK? What is the scenario for that?

**Comment by Youssef1313** [comment]
Yes, running outside of the SDK. I'll try to get more info for "what is the scenario for that" question as I don't have enough context on it.
